packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/project-has-auth.ts

```diff
@@ -6,20 +6,22 @@ import { messages } from '../assets/string-maps';
  * Checks if auth already exists in the project and prints a warning if so.
  * Returns true if auth already exists, false otherwise
  */
-export const projectHasAuth = (context: $TSContext): boolean => {
+export const projectHasAuth = (context: $TSContext, showWarning = true): boolean => {
   const meta = stateManager.getMeta(undefined, { throwIfNotExist: false });
   const existingAuthResources: [string, $TSObject][] = Object.entries(meta?.auth || {});
 
   if (existingAuthResources.length > 0) {
-    if (checkAuthIsImported(existingAuthResources)) {
-      // determine which command will make the help message useful, defaults to 'import'
-      const commandVerb = context?.input?.command && context.input.command !== 'update' ? context.input.command : 'import';
-      printer.warn(
-        'Auth has already been imported to this project and cannot be modified from the CLI. ' +
-          `To modify, run "amplify remove auth" to unlink the imported auth resource. Then run "amplify ${commandVerb} auth".`,
-      );
-    } else {
-      printer.warn(messages.authExists);
+    if (showWarning) {
+      if (checkAuthIsImported(existingAuthResources)) {
+        // determine which command will make the help message useful, defaults to 'import'
+        const commandVerb = context?.input?.command && context.input.command !== 'update' ? context.input.command : 'import';
+        printer.warn(
+          'Auth has already been imported to this project and cannot be modified from the CLI. ' +
+            `To modify, run "amplify remove auth" to unlink the imported auth resource. Then run "amplify ${commandVerb} auth".`,
+        );
+      } else {
+        printer.warn(messages.authExists);
+      }
```

Comment:
So , projectHasAuth works this way , whenever authExists it print warning based on if auth is imported or not. and this function is used consistently in `amplify update auth` and `amplify import auth ` . For Simplicity and to avoid duplication like this : https://github.com/aws-amplify/amplify-cli/blob/4e3891b3f5c5a36a3886d1f3ff7f9107d0e3cd2b/packages/amplify-category-geo/src/provider-controllers/index.ts#L77 , I added a flag to stop the warning .

Shall I create another function to check authExists instead of changing default behaviour of this function?

Reply:
The `projectHasAuth` should never been written in this way. So it's opportunity to make it better.
Fortunatelly `projectHasAuth` has only 6 callers across non-test codebase. So I'd just extract the warning logic into separate function and put in those existing blocks. It's a simple refactoring to do.

I would not want to see another clone of `projectHasAuth` with sightly different behavior.